src/app/Login/BasicAuthForm.tsx

```diff
@@ -35,15 +35,15 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import * as React from 'react';
+import { AuthMethod } from '@app/Shared/Services/Login.service';
 import { ServiceContext } from '@app/Shared/Services/Services';
 import { ActionGroup, Button, Checkbox, Form, FormGroup, Text, TextInput, TextVariants } from '@patternfly/react-core';
+import { Base64 } from 'js-base64';
+import * as React from 'react';
 import { map } from 'rxjs/operators';
 import { FormProps } from './FormProps';
-import { Base64 } from 'js-base64';
-import { AuthMethod } from '@app/Shared/Services/Login.service';
 
-export const BasicAuthForm: React.FunctionComponent<FormProps> = (props) => {
+export const BasicAuthForm: React.FunctionComponent<FormProps> = ({ onSubmit }) => {
```

Comment:
Do we need `{, ...props}`? I notice we are using this notions in many places. Is this the recommended way?

Reply:
I just notice that it seems when callback and memo needs function props as dependencies, eslint needs `props` itself to also be in the dependency list, and then having `props` in the dep list doesn't need the `props.[functionProp]` to be in the list either. However, non-function properties don't need this, and we directly list `props.whateverFieldProp` as a hook dependency. This can be done this way so that the props don't have to be destructured if the props itself isn't large (e.g. 10 props).  

For the future though, yes I will add `...props` in case for future addional `FormProps`, but I did it this way in this file only cause I knew the only prop present was the `onSubmit` function prop.

Reply:
Hmm... actually eslint will complain since `...props` won't be used. I will keep it as it is.